### PR TITLE
[Distributed] add in_subgraph on DistGraph.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
The distributed version of in_subgraph.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/sampling.py [hunk 1]
```diff
@@ -12,10 +13,10 @@
 __all__ = ['sample_neighbors']
 
 SAMPLING_SERVICE_ID = 6657
+INSUBGRAPH_SERVICE_ID = 6658
 
-
-class SamplingResponse(Response):
-    """Sampling Response"""
+class SubgraphResponse(Response):
```
Comment (VoVAllen) [inline python/dgl/distributed/sampling.py:18]: Why is this under sampling.py? Seems we need a new name for this file or move it to a separate one

File: tests/distributed/test_distributed_sampling.py [hunk 2]
```diff
@@ -132,10 +132,69 @@ def test_rpc_sampling_shuffle():
...
+    assert np.array_equal(
+        F.asnumpy(sampled_graph.edata[dgl.EID]), F.asnumpy(eids))
+
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+@unittest.skipIf(dgl.backend.backend_name == 'tensorflow', reason='Not support tensorflow for now')
+def test_rpc_in_subgraph():
+    import tempfile
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmpdirname = "/tmp/sampling"
+    check_rpc_in_subgraph(Path(tmpdirname), 2)
```
Comment (VoVAllen) [inline tests/distributed/test_distributed_sampling.py:191]:
```suggestion
    with tempfile.TemporaryDirectory() as tmpdirname:
        check_rpc_in_subgraph(Path(tmpdirname), 2)
```

File: tests/distributed/test_distributed_sampling.py [hunk 3]
```diff
@@ -132,10 +132,69 @@ def test_rpc_sampling_shuffle():
...
+    src, dst = sampled_graph.edges()
+    g = dgl.as_heterograph(g)
+    assert sampled_graph.number_of_nodes() == g.number_of_nodes()
+    subg1 = dgl.in_subgraph(g, nodes)
+    src1, dst1 = subg1.edges()
+    assert np.all(np.sort(F.asnumpy(src)) == np.sort(F.asnumpy(src1)))
+    assert np.all(np.sort(F.asnumpy(dst)) == np.sort(F.asnumpy(dst1)))
+    eids = g.edge_ids(src, dst)
+    assert np.array_equal(
+        F.asnumpy(sampled_graph.edata[dgl.EID]), F.asnumpy(eids))
```
Comment (VoVAllen) [inline tests/distributed/test_distributed_sampling.py:183]: Better to compare the result with dgl.in_subgraph

File: tests/distributed/test_distributed_sampling.py [hunk 4]
```diff
@@ -132,10 +132,69 @@ def test_rpc_sampling_shuffle():
         check_rpc_sampling_shuffle(Path(tmpdirname), 2)
         check_rpc_sampling_shuffle(Path(tmpdirname), 1)
 
+def start_in_subgraph_client(rank, tmpdir, disable_shared_mem, nodes):
+    import dgl
+    gpb = None
+    if disable_shared_mem:
+        _, _, _, gpb = load_partition(tmpdir / 'test_in_subgraph.json', rank)
+    dist_graph = DistGraph("rpc_sampling_ip_config.txt", "test_in_subgraph", gpb=gpb)
```
Comment (VoVAllen) [inline tests/distributed/test_distributed_sampling.py:140]: change "rpc_sampling_ip_config.txt" And I would prefer put them in separate files for simplicity

File: tests/distributed/test_distributed_sampling.py (hunk 3 above)
Comment (zheng-da) [reply tests/distributed/test_distributed_sampling.py:183]: the order of the edges in the subgraphs isn't the same. we have to sort them, in which case, we cannot check if the edge Ids are associated with the right endpoint nodes. this check can check the correctness of the results as well.

File: python/dgl/distributed/sampling.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/distributed/sampling.py:18]: the code of `in_subgraph` has a large overlap with sampling. we could rename the file. what is a better name?

**Pull request merged**